src/lib/tecs/TECS.cpp

```diff
@@ -172,14 +172,20 @@ void TECSAltitudeReferenceModel::update(const float dt, const AltitudeReferenceS
 	bool control_altitude = true;
 	float altitude_setpoint = setpoint.alt;
 
-	if (PX4_ISFINITE(setpoint.alt_rate)) {
+	if (!PX4_ISFINITE(setpoint.alt) && !PX4_ISFINITE(setpoint.alt_rate)) {
+		// neither altitude nor altitude rate is set - keep previous altitude setpoint
```

Comment:
@sfuhrer This comment is not really correct is it? It should say that we reset to the current altitude.

Reply:
true, adapted it.